demo.ipynb

```diff
@@ -32,15 +31,36 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
-   "metadata": {
-    "collapsed": false
-   },
-   "outputs": [],
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(60000, 28, 28)"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
     "(x_train, y_train), (x_test, y_test) = mnist.load_data()\n",
-    "x_train = x_train.reshape(x_train.shape[0], 1, 28, 28).astype('float32') / 255\n",
-    "x_test = x_test.reshape(x_test.shape[0], 1, 28, 28).astype('float32') / 255\n",
+    "x_train.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set image format so that it's compatable with backend\n",
+    "image_format = (28, 28, 1) if K.image_data_format() is 'channels_last' else (1, 28, 28 )\n",
```

Comment:
cool I shall update to `channels_last` this as you say. 

Re BN keras default axis for bn is -1 (channels_last) so if you're using Theano  and channels_first config it should be set to 1 to normalize along the feature axis.

see [docstring at BatchNormalization](https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py)